src/Redux/Home/HomePageSlice.js

```diff
@@ -0,0 +1,70 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const urlBase = 'https://api.coingecko.com/api/v3/coins/';
+
+export const fetchCategories = createAsyncThunk('homepage/fetchCategories', async () => {
+  try {
+    const response = await fetch(urlBase);
+    const data = await response.json();
+
+    return data;
+  } catch (error) {
+    throw new Error('Failed to fetch categories');
+  }
+});
+
+export const fetchCategory = createAsyncThunk(
+  'homepage/fetchCategory',
+  async ({ rejectWithValue }) => {
+    try {
+      const response = await fetch(urlBase);
+      const data = await response.json();
+
+      return data;
+    } catch (error) {
+      return rejectWithValue(error.message);
+    }
+  },
+);
+
+const homepageSlice = createSlice({
+  name: 'homepage',
+  initialState: {
+    filter: '',
+    categories: [],
+    loading: false,
+    error: null,
+  },
+  reducers: {
+    setFilter: (state, action) => {
+      state.filter = action.payload;
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchCategories.pending, (state) => {
+        state.loading = true;
+        state.error = null;
+      })
+      .addCase(fetchCategories.fulfilled, (state, action) => {
+        state.loading = false;
+        state.categories = action.payload;
+      })
+      .addCase(fetchCategories.rejected, (state, action) => {
+        state.loading = false;
+        state.error = action.payload;
+      })
+      .addCase(fetchCategory.fulfilled, (state, action) => {
+        const categoryIndex = state.categories.findIndex(
+          (category) => category.id === action.payload.id,
+        );
+        if (categoryIndex !== -1) {
+          state.categories[categoryIndex] = action.payload;
+        }
+      });
+  },
+});
```

Comment:
- [ ] Hello! 🎓 I'd like to emphasize the importance of avoiding direct state changes, even when using createSlice with Immer under the hood. While Immer simplifies state mutations, it's still a good practice to work with Redux state immutably.

Instead of directly modifying state properties, consider using spreading or concatenation to create new state objects. This helps maintain code clarity, predictability, and ensures that you're following best practices for Redux.

For instance, when updating an array in your state, use methods like concat or the spread operator (...) to create a new array with the changes. When updating an object, create a new object with the spread operator. This approach not only aligns with Redux principles but also makes your code more readable and easier to debug.

Remember, one of the core principles of Redux is immutability, and it's a good habit to follow even when tools like createSlice abstract some of the complexity. It'll lead to cleaner, more maintainable code in the long run. 😊🧹

So, keep up the great work, and remember that coding practices that promote immutability contribute to more robust and maintainable applications! 🚀📚